app/src/main/java/com/amaze/filemanager/asynchronous/services/ProgressiveServiceAbstract.java

```diff
@@ -0,0 +1,277 @@
+package com.amaze.filemanager.asynchronous.services;
+
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.app.NotificationCompat;
+import android.text.format.Formatter;
+import android.util.Log;
+
+import com.amaze.filemanager.R;
+import com.amaze.filemanager.activities.MainActivity;
+import com.amaze.filemanager.filesystem.HybridFile;
+import com.amaze.filemanager.fragments.ProcessViewerFragment;
+import com.amaze.filemanager.ui.notifications.NotificationConstants;
+import com.amaze.filemanager.utils.DatapointParcelable;
+import com.amaze.filemanager.utils.ProgressHandler;
+import com.amaze.filemanager.utils.ServiceWatcherUtil;
+
+import java.util.ArrayList;
+
+/**
+ * @author Emmanuel Messulam <emmanuelbendavid@gmail.com>
+ *         on 28/11/2017, at 19:32.
+ */
+
+public abstract class ProgressiveServiceAbstract extends Service implements ServiceWatcherUtil.ServiceWatcherInteractionInterface {
```

Comment:
Why rename to ...Abstract? There is no additional useful info added... Also, renaming all abstract classes to ...Abstract makes no sense.

Reply:
It helps when we're taking a quick look. Just like 'MainActivity' or 'CopyService'.

Reply:
Are you using Android Studio, because it changes the icon for the class. Also, this makes the name even longer. @TranceLove thoughts?

Reply:
If I'm coding in Java I'd call this `AbstractProgressiveService`; some may call it `BaseProgressiveService` or `ProgressiveServiceBase`... however name it `...Abstract` is a bit awkward to me.

> `ServiceWatcherUtil.ServiceWatcherInteractionInterface`

I'd prefer a shorter name for this. I'm used to use long variable/method names so the code can document itself when everyone are too lazy to write docs (some colleagues hate me doing this ;)); but cases like this doesn't fulfill such argument.

Reply:
Alright, will rename it to 'AbstractProgressiveService'